### PR TITLE
feat(frontend): support SET TIME ZONE

### LINKED ISSUE #8551: Support `SET TIME ZONE LOCAL` syntax
### Is your feature request related to a problem? Please describe.

Currently, we've supported the following syntax to set timezone.

```sql
SET timezone TO 'Asia/Shanghai';
```

And [PostgreSQL](https://www.postgresql.org/docs/current/sql-set.html) also supports the following syntax:

```sql
SET [ SESSION | LOCAL ] TIME ZONE { value | 'value' | LOCAL | DEFAULT }
```

The new syntax allow we set the timezone to the server local timezone, which is useful in testing.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See issue for the detail

Closes #8551 

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

- SQL commands, functions, and operators

### Release note

* Support `SET TIME ZONE` syntax

</details>

### CODE REVIEW

**Comment by xiangjinwu** [comment]
> how to get the local timezone as a string

What about https://docs.rs/iana-time-zone/0.1.53/iana_time_zone/ ?

File: src/sqlparser/tests/testdata/set.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# This file is automatically generated. See `src/sqlparser/test_runner/src/bin/apply.rs` for more information.
+- input: SET TIME ZONE LOCAL
+  formatted_sql: SET TIME ZONE LOCAL
+- input: SET TIME ZONE DEFAULT
+  formatted_sql: SET TIME ZONE DEFAULT
+- input: SET TIME ZONE "Asia/Shanghai"
+  formatted_sql: SET TIME ZONE "Asia/Shanghai"
+- input: SET TIME ZONE 'Asia/Shanghai'
```
Comment (TennyZhuang) [inline src/sqlparser/tests/testdata/set.yaml:8]: The error is unexpected, but I don't want to fix it now. Will submit an issue after the PR got merged.

File: src/frontend/src/handler/variable.rs [hunk 2]
```diff
@@ -48,6 +48,24 @@ pub fn handle_set(
...
+    handler_args: HandlerArgs,
+    value: SetTimeZoneValue,
+) -> Result<RwPgResponse> {
+    let tz_info = match value {
+        SetTimeZoneValue::Local => iana_time_zone::get_timezone()
+            .map_err(|e| ErrorCode::InternalError(format!("Failed to get local time zone: {}", e))),
+        SetTimeZoneValue::Default => Ok("UTC".to_string()),
+        SetTimeZoneValue::Literal(Value::DoubleQuotedString(s))
+        | SetTimeZoneValue::Literal(Value::SingleQuotedString(s)) => Ok(s),
+        _ => Ok(value.to_string()),
```
Comment (xiangjinwu) [inline src/frontend/src/handler/variable.rs:61]: `DoubleQuotedString` is not a PostgreSQL thing. It is considered an identifier instead:
```
dev=> set time zone "Asia/Shanghai";
ERROR:  QueryError: Invalid value ["\"Asia/Shanghai\""] for ["TIMEZONE"]
```
The `to_string` in wildcard match arm includes quotes as part of value unexpectedly. We should use `Ident::real_value` here.

**Review by xiangjinwu — COMMENTED** [review]
rest lgtm

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**